### PR TITLE
Improve collect_default_updates

### PR DESCRIPTION
**Pull request by ricardoV94**
This is part of some work to make CustomDist more powerful, specially for timeseries distributions. 

Immediate changes:

* It works with nested RNGs
* It raises error if RNG used in SymbolicRandomVariable is not given an update
* It raises warning if same RNG is used in multiple nodes

### CODE REVIEW

File: pymc/pytensorf.py [hunk 1]
```diff
@@ -1000,42 +1000,85 @@ def reseed_rngs(
...
-        and var not in inputs
+    for shared_rng in (
+        inp
+        for inp in graph_inputs(outputs, blockers=inputs)
+        # TODO: Test this directly
+        if (
+            (not must_be_shared or isinstance(inp, SharedVariable))
+            and isinstance(inp.type, RandomType)
+        )
     ):
```
Comment (michaelosthege) [inline pymc/pytensorf.py]: Extracting this iterator into a local variable would make it easier to read.

Extracting it into a function could make it easier to test too?

**Review by michaelosthege — APPROVED** [review]

File: pymc/pytensorf.py (hunk 1 above)
Comment (ricardoV94) [reply pymc/pytensorf.py]: Removed the TODO, by adding a direct test. I think this test makes the behavior of the function pretty clear, including the hard to read iterator?

https://github.com/pymc-devs/pymc/blob/d8c57f4af5e272a0d924c76caae625d6f1d0fd1f/tests/test_pytensorf.py#L542-L553

Comment (michaelosthege) [reply pymc/pytensorf.py]: I still find the iterator hard to read, but proceed on your discretion

Comment (ricardoV94) [reply pymc/pytensorf.py]: I am still being stubborn, but added a comment and renamed the iterator variable to be a bit more obvious. Will merge when tests pass again :)

https://github.com/pymc-devs/pymc/blob/87c4fe7f6cce8a90d58fc890be63fa36b876e92f/pymc/pytensorf.py#L1062-L1073

Thanks!

**Pull request merged**